AppCenterDistribute/AppCenterDistributeTests/MSDistributeTests.m

```diff
@@ -2298,6 +2319,66 @@ - (void)testReportReleaseInstallForPublicGroupWhenReleaseHashesMatch {
   assertThat(reportingParametersForUpdatedRelease[kMSURLQueryDownloadedReleaseIdKey], equalTo(@1));
 }
 
+- (void)testCheckLatestFirstNewDistributionGroupId {
+
+  // If
+  NSString *distributionGroupId = @"GROUP-ID";
+  id distributeMock = OCMPartialMock(self.sut);
+  [distributeMock setUpdateTrack:MSUpdateTrackPrivate];
+
+  // Mock the HTTP client.
+  id httpClientMock = OCMPartialMock([MSHttpClient new]);
+  id httpClientClassMock = OCMClassMock([MSHttpClient class]);
+  OCMStub([httpClientClassMock alloc]).andReturn(httpClientMock);
+  OCMStub([httpClientMock initWithMaxHttpConnectionsPerHost:4]).andReturn(httpClientMock);
+  self.sut.appSecret = kMSTestAppSecret;
+  id reachabilityMock = OCMClassMock([MS_Reachability class]);
+  OCMStub([reachabilityMock reachabilityForInternetConnection]).andReturn(reachabilityMock);
+  OCMStub([reachabilityMock currentReachabilityStatus]).andReturn(ReachableViaWiFi);
+
+  // Mock the http client calls.
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Request completed."];
+  OCMStub([httpClientMock requestCompletedWithHttpCall:OCMOCK_ANY data:OCMOCK_ANY response:OCMOCK_ANY error:OCMOCK_ANY])
+      .andForwardToRealObject()
+      .andDo(^(__unused NSInvocation *invocation) {
+        [expectation fulfill];
+      });
+  OCMClassMock([MSHttpCall class]);
```

Comment:
I don't think you need this `MSHttpCall` mock in this test context.

Reply:
Don't I need to to reject 404s from the call?

Reply:
This isn't rejecting a call but simulating retry on 404 error.

Reply:
Pushed.